src/java/voldemort/server/protocol/admin/AdminServiceRequestHandler.java

```diff
@@ -1943,10 +1943,25 @@ private Message handleDisableStoreVersion(VAdminProto.DisableStoreVersionRequest
         }
 
         if (response.getDisableSuccess()) {
-            // Then we also want to put the server in offline mode
+            // Voldemort Client can detect and handle failure only at a server
+            // level. Disabling a store inside Server is not possible . Client
+            // creates connection to the server and shares the same connection
+            // for multiple stores. Client failure detection is based on this
+            // connection error. So entire server will be marked as offline for
+            // now.
+
+            boolean roFetchState = metadataStore.getReadOnlyFetchEnabledUnlocked();
+
             VAdminProto.SetOfflineStateRequest offlineStateRequest =
                     VAdminProto.SetOfflineStateRequest.newBuilder().setOfflineMode(true).build();
             handleSetOfflineState(offlineStateRequest);
```

Comment:
I believe, that will work, but I need to run it through the SREs to see where all they use this now.

Some times when they have to take out a node due to degraded SSDs, it would mean running two commands instead of one. 

If Possible, I would like to leave this as it is for now.